luna/gateware/debug/ila.py

```diff
@@ -77,7 +78,7 @@ def __init__(self, *, signals, sample_depth, domain="sync", sample_rate=60e6, sa
         #
         # Create a backing store for our samples.
         #
-        self.mem = Memory(width=self.sample_width, depth=sample_depth, name="ila_buffer")
+        self.mem = Memory(shape=Shape(self.sample_width), depth=sample_depth, init=[])
```

Comment:
It's not necessary to explicitly construct a `Shape` here, any [shape-like](https://amaranth-lang.org/docs/amaranth/v0.5.3/guide.html#shape-casting) can be used, and since that includes integers, you can simply just pass `shape=self.sample_width`. `shape` is effectively a superset of the old `width` argument.

This goes  for the other `Memory` instances in this commit as well.

Reply:
Fixed, thanks!